Code/docker.js

```diff
@@ -1704,6 +1707,37 @@ define([
                 }
             }
 
+            // on keyup for .wcFrameButtonBar > .wcFrameButton
+            function __onKeyUpPanelButton(e) {
+                if(e.which === 13 || e.which === 32) {
+                    $('body').removeClass('wcDisableSelection');
+                    for (var i = 0; i < self._frameList.length; ++i) {
+                        var frame = self._frameList[i];
+                        if(frame.$frame.find(e.target).length > 0) {
+                            if(frame.$close[0] == e.target) {
+                                self.__closePanel(frame.panel());
+                                return;
+                            } else if(frame.$tabLeft[0] == e.target) {
+                                frame._tabScrollPos -= frame.$tabBar.width() / 2;
+                                if (frame._tabScrollPos < 0) {
+                                    frame._tabScrollPos = 0;
+                                }
+                                frame.__updateTabs();
```

Comment:
Please do these operations asynchronously using setTimeout(..) to avoid operations to be done on event/main thread.